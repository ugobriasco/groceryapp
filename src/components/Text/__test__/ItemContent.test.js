import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import Icon from  'react-native-vector-icons/MaterialIcons';

import {ItemContent, styles} from '../';

describe('ItemContent', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<ItemContent />).toJSON();
	expect(tree).toBeTruthy();

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('size styling props', () => {
		it('should render title and subtitles with a given color as defined', () => {
			const rendered = renderer.create(
				<ItemContent 
					title="foo" 
					subtitle1="foo1" 
					subtitle2="foo2" 
					color="red"
				/>);
			expect(rendered).toMatchSnapshot();	
		});
	});
});



