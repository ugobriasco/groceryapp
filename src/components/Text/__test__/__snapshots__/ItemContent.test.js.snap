// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemContent size styling props should render title and subtitles with a given color as defined 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        Object {
          "color": "red",
        },
      ]
    }
  >
    foo
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        Object {
          "color": "red",
        },
      ]
    }
  >
    foo1
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        Object {
          "color": "red",
        },
      ]
    }
  >
    foo2
  </Text>
</View>
`;
